### PR TITLE
Allow plugins to get the data they set

### PR DESCRIPTION
**Pull request by codebrainz**

### CODE REVIEW

**Comment by codebrainz** [comment]
Ammended to add `@see geany_plugin_set_data()` reference.

File: src/pluginutils.c [hunk 1]
```diff
@@ -554,6 +554,24 @@ void plugin_builder_connect_signals(GeanyPlugin *plugin,
...
+ * @return The data corresponding to the plugin or @c NULL if none set.
+ *
+ * @since 1.29 (API 230)
+ *
+ * @see geany_plugin_set_data()
+ */
+gpointer geany_plugin_get_data(const GeanyPlugin *plugin)
+{
+	g_return_val_if_fail (plugin != NULL, NULL);
+	g_return_val_if_fail (PLUGIN_LOADED_OK (plugin->priv), NULL);
```
Comment (elextr) [inline src/pluginutils.c]: These should be normal `if(..)return val` because they should not be removable by compile option as the `g_...` checks are.

Comment (codebrainz) [inline src/pluginutils.c]: They check programming errors and so should be removed for non-debug builds.

Comment (elextr) [inline src/pluginutils.c]: Indeed, but plugins are external code, and can pass bad parameters at any time. All functions in the API should be permanently checked.  (yes Geany is not consistent about that, but you can have the honour of being the first to get it right :)

Comment (codebrainz) [inline src/pluginutils.c]: Look at [the function below](https://github.com/geany/geany/pull/1234/commits/84720babfbce2642289a8c038137402533e5d2c5#diff-68052c9feb0f8c0562bf6f40b8b19669R602) this one. I copied the pre-conditions but added an additional one for the `plugin != NULL` for good measure.

**Comment by kugel-** [comment]
@codebrainz @elextr this can be closed (was commited to master as 7261742f8169b19fd5ac7f10a0845a684287e2d9)

**Pull request closed without merging**